deploy/pulumi/providers/azure.py

```diff
@@ -0,0 +1,198 @@
+import pulumi
+import pulumi_azure as azure
+import pulumi_azuread as azuread
+from pulumi_azure.containerservice import (
+    KubernetesCluster,
+    KubernetesClusterDefaultNodePoolArgs,
+    KubernetesClusterServicePrincipalArgs,
+    KubernetesClusterNetworkProfileArgs,
+)
+from typing import Union
+from .base import Cluster, cluster_types, Database, database_types, get_password
+
+
+class AzureBase:
+    """
+    Base Azure resources required for
+    """
+
+    _created: bool = False
+
+    # Created resources
+    resource_group: azure.core.ResourceGroup
+    app: azuread.Application
+    service_principal: azuread.ServicePrincipal
+    service_principal_pwd: azuread.ServicePrincipalPassword
+    network: Union[
+        azure.network.VirtualNetwork, azure.network.AwaitableGetVirtualNetworkResult
+    ]
+    private_subnet: azure.network.Subnet
+
+    def __init__(self):
+        self.config = pulumi.Config("prefect-server-azure")
+
+    def create_if_not_created(self):
+        if not self._created:
+            self.create()
+            self._created = True
+
+    def create(self):
+        self.resource_group = azure.core.ResourceGroup("prefect-server-rg-")
+
+        self.app = azuread.Application("prefect-server-app")
+
+        self.service_principal = azuread.ServicePrincipal(
+            "prefect-server-app-sp",
+            application_id=self.app.application_id,
+        )
+
+        self.service_principal_pwd = azuread.ServicePrincipalPassword(
+            "prefect-server-app-sp-pwd",
+            service_principal_id=self.service_principal.id,
+            end_date="2099-01-01T00:00:00Z",
+            value=get_password("service-principal-password", self.config),
+        )
+
+        existing_network = pulumi.Config.get("existing-network-name")
```

Comment:
```suggestion
        existing_network = self.config.get("existing-network-name")
```

Reply:
`self.config` trickily points to `prefect-server-azure` whilst `pulumi.Config` points to `prefect-server`